c/cmake_modules/BuildUtils.cmake

```diff
@@ -240,6 +240,21 @@ function(ADD_ARROW_LIB LIB_NAME)
                                      OUTPUT_NAME ${LIB_NAME}
                                      VERSION "${ADBC_FULL_SO_VERSION}"
                                      SOVERSION "${ADBC_SO_VERSION}")
+                                     
+    if(MSVC)
+        # Binaries generated on Windows need file version information, otherwise when the binary is part of a Windows installer
+        # the installer won't know to update a previously installed version.
+        set(VERSION_RC_TEMPLATE "${CMAKE_SOURCE_DIR}/cmake_modules/version.rc.in")
```

Comment:
How about moving `version.rc.in` to `c/` from `c/cmake_modules/`?
We use `cmake_modules/` for `.cmake` that are included by `include(...)` in `CMakeLists.txt`/`*.cmake`.

Reply:
Moved to c/